fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryAbstractDataModificationPlan.java

```diff
@@ -126,6 +126,7 @@ public abstract class RecordQueryAbstractDataModificationPlan implements RecordQ
     private final Supplier<Integer> planHashForContinuationSupplier;
     @Nonnull
     private final Supplier<Integer> planHashForWithoutLiteralsSupplier;
+    private boolean dryRun;
```

Comment:
I have a suggestion for refactoring that I left in `RecordQueryUpdatePlan` that would allow this field to be removed. In general, having mutable state in the plan is dangerous, because plan objects can be cached and re-used across threads, and if different threads are run with different values of the dry-run flag, the value could get corrupted.